### PR TITLE
Documentation in place for BlazeBuilder

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
Just me sketching stuff out while I looked over it today.

### CODE REVIEW

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 1]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
 import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
 
+/**
+ * BlazeBuilder is the component for the builder pattern aggregating
+ * different components to finally serve requests.
+ *
+ * Variables:
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: I think if you put a `@param` annotation on these, it'll do the right thing for scaladoc.

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 2]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * enableWebsockets: Enables Websocket Support
+ * sslBits: If present enables secure communication to the server using the 
+ *    sslContext
+ * isHttp2Enabled: Whether or not to enable Http2 Server Features
+ * maxRequestLineLength: Maximum request line to parse
+ * maxHeadersLen: Maximum data that composes the headers
+ * serviceMounts: The services that are mounted on this server to serve.
+ * serviceErrorHandler: The last resort to recover and generate a response
+ *    this is necessary to recovery totality from the error condition.
+ * banner: Pretty log to display on server start.
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: Worth mentioning Nil disables this

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 3]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+/**
+ * BlazeBuilder is the component for the builder pattern aggregating
+ * different components to finally serve requests.
+ *
+ * Variables: 
+ * socketAddress: Is the Socket Address the server will be mounted at
+ * executionContext: Is the Execution Context the underlying blaze futures
+ *   will be executed upon.
+ * idleTimeout: Period of Time a connection can remain idle before the
+ *    connection is timed out and disconnected.
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: `Duration.Inf` disables it.

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 4]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * isNio2: Whether or not to use NIO2 or NIO1 Socket Server Group
+ * connectorPoolSize: Number of worker threads for the new Socket Server Group
+ * bufferSize: Buffer size to use for IO operations
+ * enableWebsockets: Enables Websocket Support
+ * sslBits: If present enables secure communication to the server using the
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: s/If present/If defined,/

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 5]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * idleTimeout: Period of Time a connection can remain idle before the
+ *    connection is timed out and disconnected.
+ * isNio2: Whether or not to use NIO2 or NIO1 Socket Server Group
+ * connectorPoolSize: Number of worker threads for the new Socket Server Group
+ * bufferSize: Buffer size to use for IO operations
+ * enableWebsockets: Enables Websocket Support
+ * sslBits: If present enables secure communication to the server using the 
+ *    sslContext
+ * isHttp2Enabled: Whether or not to enable Http2 Server Features
+ * maxRequestLineLength: Maximum request line to parse
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: Mention that it's a `400 Bad Request` if exceeded?

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 6]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * maxHeadersLen: Maximum data that composes the headers
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: Mention that it's a `400 Bad Request` if exceeded?

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 7]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * serviceMounts: The services that are mounted on this server to serve.
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: Mention that they get assembled into a `Router`, with longest prefix winning?

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala [hunk 8]
```diff
@@ -22,6 +22,31 @@ import scala.collection.immutable
...
+ * serviceErrorHandler: The last resort to recover and generate a response
+ *    this is necessary to recovery totality from the error condition.
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeBuilder.scala]: necessary to recover totally

**Comment by ChristopherDavenport** [comment]
Closing in favor of #1775

**Pull request closed without merging**